xxx-uv_link_t.md

```diff
@@ -0,0 +1,67 @@
+| Title  | Port core to uv_link_t      |
+|--------|-----------------------------|
+| Author | @indutny                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-03                  |
+
+## 1. Proposal
+
+I propose to replace `StreamBase` with [`uv_link_t`][0], and port existing
+classes that inherit from `StreamBase`.
```

Comment:
The conflict was due to the use of `vtable` and `virtual` methods in `StreamBase`. `uv_link_t` is a C project, there won't be any such conflict.

In fact, I believe that exposing an External getter for the pointer to internal `uv_link_t` is enough to get things running. This getter can be a per-class, or could be shared, but the inheritance won't be a roadblock for this.